File: src/main/java/net/dontdrinkandroot/gitki/wicket/event/DirectoryEvent.java
package net.dontdrinkandroot.gitki.wicket.event;

import net.dontdrinkandroot.gitki.model.DirectoryPath;
import org.apache.wicket.ajax.AjaxRequestTarget;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class DirectoryEvent extends AbstractEvent
{
    private DirectoryPath directoryPath;

    public DirectoryEvent(DirectoryPath directoryPath)
    {
        this.directoryPath = directoryPath;
    }

    public DirectoryEvent(DirectoryPath directoryPath, AjaxRequestTarget target)
    {
        super(target);
        this.directoryPath = directoryPath;
    }

    public DirectoryPath getDirectoryPath()
    {
        return this.directoryPath;
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/event/DirectoryEvent.kt
package net.dontdrinkandroot.gitki.wicket.event

import net.dontdrinkandroot.gitki.model.DirectoryPath
import org.apache.wicket.ajax.AjaxRequestTarget

open class DirectoryEvent : AbstractEvent {

    var directoryPath: DirectoryPath
        private set

    constructor(directoryPath: DirectoryPath) {
        this.directoryPath = directoryPath
    }

    constructor(directoryPath: DirectoryPath, target: AjaxRequestTarget?) : super(target) {
        this.directoryPath = directoryPath
    }
}